decorator.rb

```diff
@@ -0,0 +1,14 @@
+require './nameable'
+
+# class decorator
+class Decorator < Nameable
+  def initialize(nameable)
+    @nameable = nameable
+    super()
+  end
+
+  def correct_name
+    super()
+    @nameable
+  end
```

Comment:
- [ ] this method should return the result of the correct_name method of the @namable i.e 
```
 def correct_name
      @nameable.correct_name
    end
```